react/features/toolbox/components/web/AudioSettingsButton.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+import React, { Component } from 'react';
+
+import AudioMuteButton from '../AudioMuteButton';
+import { hasAvailableDevices } from '../../../base/devices';
+import { IconArrowDown } from '../../../base/icons';
+import JitsiMeetJS from '../../../base/lib-jitsi-meet/_';
+import { ToolboxButtonWithIcon } from '../../../base/toolbox';
+import { connect } from '../../../base/redux';
+
+import { AudioSettingsPopup, toggleAudioSettings } from '../../../settings';
+
+type Props = {
+
+    /**
+     * Click handler for the small icon. Opens audio options.
+     */
+    onAudioOptionsClick: Function,
+
+    /**
+     * If the user has audio input or audio output devices.
+     */
+    hasDevices: boolean,
+
+    /**
+     * Flag controlling the visibility of the button.
+     */
+    visible: boolean,
+};
+
+type State = {
+
+    /**
+     * If there are permissions for audio devices.
+     */
+    hasPermissions: boolean,
+}
+
+/**
+ * Button used for audio & audio settings.
+ *
+ * @returns {ReactElement}
+ */
+class AudioSettingsButton extends Component<Props, State> {
+    /**
+     * Initializes a new {@code AudioSettingsButton} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hasPermissions: false
+        };
+    }
+
+    /**
+     * Updates device permissions.
+     *
+     * @returns {Promise<void>}
+     */
+    async _updatePermissions() {
+        const hasPermissions = await JitsiMeetJS.mediaDevices.isDevicePermissionGranted(
+            'audio',
+        );
+
+        this.setState({
+            hasPermissions
+        });
+    }
+
+    /**
+     * Implements React's {@link Component#componentDidMount}.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this._updatePermissions();
```

Comment:
This doesn't work if the user never gave permission. The component will be mounted and we'll never check again.

Reply:
I'm not sure about all other browsers, but at least in Chrome you need to reload the page for the permissions change to take effect, so it doesn't seem to be a problem. Or at least not until this would be landing on mobile.

Reply:
I've just checked and it is a problem on Firefox